README.md

```diff
@@ -36,339 +34,172 @@
+```sh
+# Check for Python == where the left and right hand sides are the same (often a bug)
+$ semgrep -e `$X==$X` --lang=py path/to/src
 
-# Run a set of rules
-$ semgrep --config=https://semgrep.dev/c/r/java.spring.security
+# Run the default rule pack with rules for many languages
+$ semgrep --config=default path/to/src
 ```
 
-All public Semgrep rules can be viewed on the [Registry](https://semgrep.dev/r), which pulls the rules from YAML files defined in the [semgrep-rules](https://github.com/returntocorp/semgrep-rules) GitHub repo.
-
-Here are some sample vulnerable repos to test on:
-* Django: [lets-be-bad-guys](https://github.com/mpirnat/lets-be-bad-guys), [django.nV](https://github.com/nVisium/django.nV)
-* Flask: [Vulnerable-Flask-App](https://github.com/we45/Vulnerable-Flask-App)
-* Java: [WebGoat](https://github.com/WebGoat/WebGoat), [OWASP Benchmark](https://github.com/OWASP/Benchmark)
-* NodeJS: [OWASP Juice Shop](https://github.com/bkimminich/juice-shop), [DevSlop Pixi](https://github.com/DevSlop/Pixi)
-* Golang: [GoVWA](https://github.com/0c34/govwa)
-
-### Writing Custom Rules
-
-One of the strengths of Semgrep is how easy it is to write rules.
-
-This makes it possible to:
-* Quickly port rules from other tools.
-* Think of an interesting code pattern, and then find instances of it in your
-    code.
-* Find code base or org-specific bugs and antipatterns - things that built-in
-    checks for existing tools won't find because they're unique to you.
-* and more!
-
-#### Simple Rules
-
-For iterating on simple patterns, you can use the `--lang` and `--pattern`
-flags.
-
-```bash
-$ semgrep --lang javascript --pattern 'eval(...)' path/to/file.js
-```
+Explore community rule packs and CI integrations at [semgrep.dev](https://semgrep.dev/packs).
 
-The `--lang` flag tells Semgrep which language you're targeting and `--pattern` is the code pattern to search for.
+## Examples
 
-#### Advanced Rules
+### Core Uses
 
-Some rules need more than one line of pattern to express. Sometimes you want to express code patterns, like: `X` must be true AND `Y` must be too, or `X` but NOT `Y`, or `X` must occur inside a block of code that `Y` matches.
+- Ban dangerous APIs [[1](https://semgrep.live/clintgibler:no-exec)]
+- Search routes and authentiation [[1](https://semgrep.live/clintgibler:spring-routes)]
+- Enforce the use secure defaults [[1](https://semgrep.dev/dlukeomalley:flask-set-cookie)]
+- Enforce project best-practices [[1](https://semgrep.dev/dlukeomalley:use-assertEqual-for-equality), [2](https://semgrep.dev/dlukeomalley:unchecked-subprocess-call)]
+- Codify project-specific knowledge [[1](https://semgrep.dev/dlukeomalley:verify-before-make)]
+- Audit security hotspots [[1](https://semgrep.live/ievans:airflow-xss), [2](https://semgrep.dev/dlukeomalley:hardcoded-credentials)]
+- Audit configuration files [[1](https://semgrep.dev/dlukeomalley:s3-arn-use)]
 
```

Comment:
IMO the [1] type of link formatting is a bit hard to use since the click targets are small and it’s a bit academic-looking. How about something like this?


### Example uses

| Use case                          | Semgrep rule                                                                                                                                                         |
| :-------------------------------- | :------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| Ban dangerous APIs                | [No exec](https://semgrep.live/clintgibler:no-exec)                                                                                                                  |
| Search routes and authentiation   | [Spring routes](https://semgrep.live/clintgibler:spring-routes)                                                                                                      |
| Enforce the use secure defaults   | [Set cookie](https://semgrep.dev/dlukeomalley:flask-set-cookie)                                                                                                      |
| Enforce project best-practices    | [Use assert](https://semgrep.dev/dlukeomalley:use-assertEqual-for-equality), [Unchecked subprocess call](https://semgrep.dev/dlukeomalley:unchecked-subprocess-call) |
| Codify project-specific knowledge | [Verify before make](https://semgrep.dev/dlukeomalley:verify-before-make)                                                                                            |
| Audit security hotspots           | [Airflow XSS](https://semgrep.live/ievans:airflow-xss), [Hardcoded credentials](https://semgrep.dev/dlukeomalley:hardcoded-credentials)                              |
| Audit configuration files         | [S3 ARN use](https://semgrep.dev/dlukeomalley:s3-arn-use)                                                                                                            |

Reply:
Not sure if GitHub placed this ^^^ comment in the right location. It was meant to be a comment on the code between lines [90, 98] and [100, 103].

Reply:
This looks fantastic. I am glad to have had my work improved upon 🙏